src/diffcalc_API/stores/mongo.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, Optional
+
+import numpy as np
+from diffcalc.hkl.calc import HklCalculation
+from diffcalc.hkl.constraints import Constraints
+from diffcalc.ub.calc import UBCalculation
+from pymongo.collection import Collection
+from pymongo.results import DeleteResult
+
+from diffcalc_API.database import database
+from diffcalc_API.errors.definitions import (
+    ALL_RESPONSES,
+    DiffcalcAPIException,
+    ErrorCodes,
+)
+from diffcalc_API.stores.protocol import HklCalcStore
+
+
+class Codes(ErrorCodes):
+    attempting_to_overwrite = 405
+    document_found = 404
+    nothing_to_delete = 404
```

Comment:
This could be an `Enum` or even `IntEnum`, which would provide the functionality of `ErrorCodes`

Reply:
The reason it inherits from ErrorCodes is so it can use ErrorCodes.all_codes() method. How would I keep this functionality if I were to do this?

Reply:
You'd probably still want an `ErrorCodes` class like so:

```python
class ErrorCodes(IntEnum):
    @classmethod
    @property
    def values(cls) -> set[str]:
        return {val.value for val in cls}
```

Allowing you to get a set of values with `ErrorCodes.values`